client/src/Components/SingularPost.js

```diff
@@ -26,24 +26,24 @@ export default function SingularPost(props) {
 
   const handlePostLikeClick = () => {
     if (postLikedBySelf) {
-      setPostLikedBySelf(false);
-      setPostLikes(postLikes - 1); // cosmetic
       // delete like
       authFetch(
         `/posts/${props.post_id}/likes`,
         { liker_id: props.author_id },
         props,
         "delete"
       );
+      setPostLikedBySelf(false);
+      setPostLikes(postLikes - 1); // cosmetic;
     } else {
-      setPostLikes(postLikes + 1); // cosmetic
-      setPostLikedBySelf(true);
       authFetch(
         `/posts/${props.post_id}/likes`,
         { liker_id: props.author_id },
         props,
         "post"
       );
```

Comment:
Actually, what I meant is this:

autchFetch(...)
.then(r=>){//check response status, change state if success}

Reply:
A good suggestion but I think this whole PR has been overwritten by the latest PR.